src/app/core/guards/parse-url-params.guard.ts

```diff
@@ -0,0 +1,22 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, CanActivate, Router, RouterStateSnapshot, UrlSegment } from '@angular/router';
+import { Phase } from '../models/phase.model';
+import { CacheRepoFromUrlService } from '../services/cache-repo-from-url.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ParseUrlParamsGuard implements CanActivate {
+  constructor(private router: Router, private cache: CacheRepoFromUrlService) {}
+
+  canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot) {
+    const org = route.params['org'];
+    const repo = route.params['repo'];
+    const repoLocation = `${org}/${repo}`;
+
+    this.cache.repoLocation = repoLocation;
+
+    this.router.navigateByUrl(Phase.issuesViewer);
+    return false;
+  }
```

Comment:
Unusual to have both side effects (caching) and navigation although definitely wrong! Perhaps it might be better to add some comments or documentation to why this was done (as it took some time for me to understand the logic here behind intercepting the repo location and then rerouting to `issuesViewer`)

Reply:
Hi, i agree with this being an unusual application of angular route guards, but I was not sure of how else to do it.
My original idea was to leave the /issuesViewer path unprotected and inside the ngOnInit() function of IssuesViewerComponent, i would cache the :org/:repo and redirect to login if needed.

But the above approach would require removing authGuard, which didnt seem ideal.

Would you happen to have better alternative approaches?

Reply:
My bad, I was missed a word here (I was supposed to type although definitely *not* wrong). This is completely fine! Sorry about that!